hax/hax/motr/delivery.py

```diff
@@ -46,6 +46,24 @@ def __init__(self):
         self.waiting_clients: Dict[HaLinkMessagePromise, Condition] = {}
         self.lock = Lock()
 
+    def check_delivery(self,
+                       promise: HaLinkMessagePromise,
+                       timeout_sec: float):
+        """
+        Check if any message in promise._ids are reported by Motr
+        as delivered. Calling function should hold the lock.
```

Comment:
[suggestion]: It would be good to mention the lock here (self.lock) in documentation.

Reply:
updated lock to self.lock